polygon/stream.go

```diff
@@ -86,6 +87,15 @@ func (s *Stream) Close() error {
 	return s.conn.Close()
 }
 
+func (s *Stream) reconnect() {
+	s.conn = openSocket()
```

Comment:
You would need `auth()` too?

Reply:
Turns out that there were a few issues that needed fixing, including that.